Lib/importlib/abc.py

```diff
@@ -410,6 +428,9 @@ def open(self, mode='r', *args, **kwargs):
 
         When opening as text, accepts encoding parameters such as those
         accepted by io.TextIOWrapper.
+
+        If the file cannot be found, FileNotFoundError should be raised,
```

Comment:
I'd probably change the terminology. The term "the file" could be more clear. Maybe something like "the object referenced by self".

Also, "open" may raise other exceptions. For example, if self references a directory and someone attempts to open it, they should expect "IsADirectoryError" to be raised. Or if the resource was on a network and the network has dropped, they might get a socket.ConnectionError. Or if it was an HTTP resource and access has been revoked, they may get an HTTPAccessDenied error. I'm not sure it's appropriate to attempt to prescribe which exceptions should or shouldn't be expected.